rclcpp/src/rclcpp/executor.cpp

```diff
@@ -569,3 +572,25 @@ Executor::get_next_executable(std::chrono::nanoseconds timeout)
   }
   return any_exec;
 }
+
+std::ostream &
+rclcpp::executor::operator << (std::ostream & os, const FutureReturnCode & future_return_code)
+{
+  return os << to_string(future_return_code);
+}
+
+std::string
+rclcpp::executor::to_string(const FutureReturnCode & future_return_code)
+{
+  using enum_type = std::underlying_type<FutureReturnCode>::type;
+  using std::string;
+  using std::to_string;
+  switch (future_return_code) {
+    case FutureReturnCode::SUCCESS:
+      return string("SUCCESS (" + to_string(static_cast<enum_type>(future_return_code)) + ")");
+    case FutureReturnCode::INTERRUPTED:
+      return string("INTERRUPTED (" + to_string(static_cast<enum_type>(future_return_code)) + ")");
+    case FutureReturnCode::TIMEOUT:
+      return string("TIMEOUT (" + to_string(static_cast<enum_type>(future_return_code)) + ")");
```

Comment:
Actually it doesn't since this is an Enum switch. The compiler will warn you if the enum is extended in the future that not all cases are covered.